widgets/HTTPToHDFS-action.json

```diff
@@ -0,0 +1,145 @@
+              "GET",
+              "POST"
+            ],
+            "default": "GET"
+          }
+        },
+        {
+          "widget-type": "textbox",
+          "label": "Request Body",
+          "name": "body"
+        },
+        {
+          "widget-type": "keyvalue",
+          "label": "Request Headers",
+          "name": "requestHeaders",
+          "widget-attributes": {
+            "showDelimiter": "false",
+            "kv-delimiter": ":",
+            "delimiter": "\n"
+          }
+        },
+        {
+          "widget-type": "select",
+          "label": "Output File Format",
+          "name": "outputFormat",
+          "widget-attributes": {
+            "values": [
+              "Text",
+              "Binary"
+            ],
+            "default": "Text"
+          }
+        },
+        {
+          "widget-type": "select",
+          "label": "Charset for Text",
+          "name": "charset",
+          "widget-attributes": {
+            "values": [
+              "UTF-8",
+              "US-ASCII",
+              "ISO-8859-1",
+              "UTF-16",
+              "UTF-16BE",
+              "UTF-16LE"
+            ],
+            "default": "UTF-8"
+          }
+        },
+        {
+          "widget-type": "select",
+          "label": "Should Follow Redirects ?",
+          "name": "followRedirects",
+          "widget-attributes": {
+            "values": [
+              "true",
+              "false"
+            ],
+            "default": "true"
+          }
+        },
+        {
+          "widget-type": "select",
+          "label": "Disable SSL Validation",
+          "name": "disableSSLValidation",
+          "widget-attributes": {
+            "values": [
+              "true",
+              "false"
+            ],
+            "default": "true"
+          }
+        },
+        {
+          "widget-type": "select",
+          "label": "Number of Retries",
+          "name": "numRetries",
+          "widget-attributes": {
+            "values": [
+              "3",
+              "0",
+              "1",
+              "2",
+              "4",
+              "5",
+              "6",
+              "7",
+              "8",
+              "9",
+              "10"
+            ],
+            "default": "3"
+          }
+        },
+        {
+          "widget-type": "textbox",
+          "label": "Connection Timeout (milliseconds)",
+          "name": "connectTimeout"
+        },
+        {
+          "widget-type": "textbox",
+          "label": "Read Timeout (milliseconds)",
+          "name": "readTimeout"
+        },
+        {
+          "widget-type": "textbox",
+          "label": "Token Key for HDFS File Path",
+          "name": "outputPath",
+          "default": "filePath"
+        },
+        {
+          "widget-type": "textbox",
+          "label": "Token Key for Response Headers",
+          "name": "responseHeaders",
+          "default": "responseHeaders"
```

Comment:
this isn't correct. the proper way for this is "widget-attributes" : { "default" : "responseHeaders" }

It's clear you didn't test this change.